Popular_Models_Guide/Llava1.5/llava_trtllm_guide.md

```diff
@@ -0,0 +1,222 @@
+to use.
+
+```bash
+docker run --rm -it --net host --shm-size=2g \
+    --ulimit memlock=-1 --ulimit stack=67108864 --gpus all \
+    -v </path/to/tensorrtllm_backend>:/tensorrtllm_backend \
+    -v </path/to/Llava1.5/repo>:/llava-1.5-7b-hf \
+    -v </path/to/engines>:/engines \
+    -v </path/to/tutorials>:/tutorials \
+    nvcr.io/nvidia/tritonserver:<xx.yy>-trtllm-python-py3
+```
+
+Alternatively, you can follow instructions
+[here](https://github.com/triton-inference-server/tensorrtllm_backend?tab=readme-ov-file#build-the-docker-container)
+to build Triton Server with Tensorrt-LLM Backend if you want
+to build a specialized container.
+
+Don't forget to allow gpu usage when you launch the container.
+
+### Create Engines for each model [skip this step if you already have engines]
+
+TensorRT-LLM requires each model to be compiled for the configuration
+you need before running. To do so, before you run your model for the first time
+on Triton Server you will need to create a TensorRT-LLM engine.
+
+Starting with [24.04 release](https://github.com/triton-inference-server/server/releases/tag/v2.45.0),
+Triton Server TensrRT-LLM container comes with
+pre-installed TensorRT-LLM package, which allows users to build engines inside
+the Triton container.
+
+Llava1.5 requires 2 engines: a TensorRT engine for visual components,
+and a TRT-LLM engine for the language components. This tutorial bases on 24.05
+release, which corresponds to `v0.9.0` version of TensorRT-LLM and
+TensorRT-LLM backend and follows [this](https://github.com/NVIDIA/TensorRT-LLM/tree/v0.9.0/examples/multimodal#llava-and-vila)
+TensorRT-LLM multi-modal guide.
+
+To generate engines, simply follow the next steps:
+
+```bash
+HF_LLAVA_MODEL=/llava-1.5-7b-hf
+UNIFIED_CKPT_PATH=/tmp/ckpt/llava/7b/
+ENGINE_DIR=/engines/llava1.5
+CONVERT_CHKPT_SCRIPT=/tensorrtllm_backend/tensorrt_llm/examples/llama/convert_checkpoint.py
+python3 ${CONVERT_CHKPT_SCRIPT} --model_dir ${HF_LLAVA_MODEL} --output_dir ${UNIFIED_CKPT_PATH} --dtype float16
+trtllm-build --checkpoint_dir ${UNIFIED_CKPT_PATH} \
+            --output_dir ${ENGINE_DIR} \
+            --gemm_plugin float16 \
+            --use_fused_mlp \
+            --max_batch_size 1 \
+            --max_input_len 2048 \
+            --max_output_len 512 \
+            --max_multimodal_len 576 # 1 (max_batch_size) * 576 (num_visual_features)
+
+python /tensorrtllm_backend/tensorrt_llm/examples/multimodal/build_visual_engine.py --model_path ${HF_LLAVA_MODEL} --model_type llava --output_dir ${ENGINE_DIR}/llava1.5
+```
+
+
+> Optional: You can check test the output of the model with `run.py`
+> located in the same llama examples folder.
+>
+>   ```bash
+>    python3 /tensorrtllm_backend/tensorrt_llm/examples/multimodal/run.py --max_new_tokens 30 --hf_model_dir ${HF_LLAVA_MODEL} --visual_engine_dir ${ENGINE_DIR} --llm_engine_dir ${ENGINE_DIR} --decoder_llm --input_text "Question: which city is this? Answer:"
```

Comment:
I'm seeing the error with file not found:
```
FileNotFoundError: [Errno 2] No such file or directory: '/engines/llava1.5/visual_encoder.engine'
```
I think the `--visual_engine_dir` should be `${ENGINE_DIR}/llava1.5`
```
root@a2826b5-lcedt:/engines/llava1.5/llava1.5# ls -l
total 613616
-rw-r--r-- 1 root root 628335908 Jun 19 00:28 visual_encoder.engine
```

Reply:
Thanks for catching this, I'll double check

Reply:
I've went through steps one more time and adjusted all paths to avoid double nested folders